clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -223,8 +224,9 @@ private RecordAppendResult tryAppend(long timestamp, byte[] key, byte[] value, C
      * Abort the batches that have been sitting in RecordAccumulator for more than the configured requestTimeout
      * due to metadata being unavailable
      */
-    public List<RecordBatch> abortExpiredBatches(int requestTimeout, long now) {
+    public List<RecordBatch> abortExpiredBatches(int requestTimeout, boolean staleMetadataNow, Metadata metadata, long now) {
```

Comment:
Why are we passing in the Metadata object?
We already have Cluster in the imports right? We can just pass in the cluster in that case right?